src/mlpack/core/optimizers/sgd/sgd.hpp

```diff
@@ -109,10 +109,24 @@ class SGD
    * starting point will be modified to store the finishing point of the
    * algorithm, and the final objective value is returned.
    *
+   * @param function Function to optimize.
    * @param iterate Starting point (will be modified).
    * @return Objective value of the final point.
    */
-  double Optimize(arma::mat& iterate);
+  double Optimize(DecomposableFunctionType& function, arma::mat& iterate);
```

Comment:
yeah that's true. We can do that when we refactor all the optimizers systematically. But for this PR it's enough.